### PR TITLE
IA-892 instance period filter

### PR DESCRIPTION
**Pull request by beygorghor**
- refactor instance filters with local state and avoid FilterComponent
- add period type, start and end period filter
- remove period reducer

Screenshot 2021-11-15 at 11 50 24
Screenshot 2021-11-15 at 12 38 35

Screenshot 2021-11-15 at 11 53 19

### CODE REVIEW

File: hat/assets/js/apps/Iaso/domains/instances/components/InstancesFiltersComponent.js [hunk 1]
```diff
@@ -136,111 +77,213 @@ const InstancesFiltersComponent = ({
...
+    const startPeriodError = useMemo(() => {
+        if (formState.startPeriod?.value && formState.periodType?.value) {
+            return !isValidPeriod(
+                formState.startPeriod.value,
+                formState.periodType.value,
+            );
+        }
+        return false;
+    }, [formState.startPeriod, formState.periodType]);
+    const endPeriodError = useMemo(() => {
```
Comment (olethanh) [inline hat/assets/js/apps/Iaso/domains/instances/components/InstancesFiltersComponent.js:102]: wouldn't it make more sense to use setFieldErrors and to set it in the handleFormChange?

File: hat/assets/js/apps/Iaso/domains/instances/constants.js [hunk 2]
```diff
@@ -26,7 +26,8 @@ const PrettyPeriod = ({ value }) => {
 };
 
 const LinkToForm = ({ formId, formName }) => {
-    const user = useSelector(state => state.users.current);
+    const user =
```
Comment (olethanh) [inline hat/assets/js/apps/Iaso/domains/instances/constants.js:29]: seems like a merge error, this broke the feature

**Comment by quang-le** [comment]
QA test in Jira: https://bluesquare.atlassian.net/browse/IA-936?atlOrigin=eyJpIjoiZWY4ODM2YjVlNmNjNDZkMmJmODVmYzcwYTY3ZTZlNTEiLCJwIjoiaiJ9

**Review by quang-le — CHANGES_REQUESTED** [review]
See [Jira ticket](https://bluesquare.atlassian.net/browse/IA-936?atlOrigin=eyJpIjoiZWY4ODM2YjVlNmNjNDZkMmJmODVmYzcwYTY3ZTZlNTEiLCJwIjoiaiJ9)

Blocking and in scope (number refers to ticket description): 6
In scope but not critical: 5
Not in scope of this PR, but related to submissions: 1,2,3,4

**Review by quang-le — APPROVED** [review]

**Pull request merged**